cryptocoins/src/app/pages/home/home.component.ts

```diff
@@ -10,28 +14,64 @@ import { CryptolistService } from '../../services/cryptolist.service';
 export class HomeComponent implements OnInit {
   constructor(private cryptoListService: CryptolistService) {}
   rowData: Coin[] = [];
-
+  searchValue: any;
+  gridApiActive: any;
+  serverSideDatasource: any;
   ngOnInit(): void {
-    this.getHttpData();
+    this.columnDefs = [
+      {
+        field: 'name',
+        width: 100,
+        editable: true,
+        filter: 'agTextColumnFilter',
+      },
+      { field: 'symbol', width: 80 },
+      {
+        field: 'price',
+        width: 80,
+        filter: 'agNumberColumnFilter',
+      },
+    ];
   }
 
-  columnDefs: ColDef[] = [
-    {
-      field: 'name',
-      width: 100,
-      editable: true,
-    },
-    { field: 'symbol', width: 80 },
-    { field: 'price', width: 80 },
-  ];
+  columnDefs: ColDef[] = [];
+  rowModelType: 'serverSide' = 'serverSide';
   handleError() {}
 
   getHttpData() {
     this.cryptoListService.getJSON().subscribe((data) => {
       this.rowData = data;
     });
   }
+
+  getData() {
+    this.cryptoListService.getJSON().subscribe((data: any) => {
+      this.serverSideDatasource = createServerSideDatasource(data.slice());
```

Comment:
Why do you need `.slice()` here?

Reply:
And why do you desided to go with creating the method `createServerSideDatasource` ?